subworkflows/local/long_reads_qc.nf

```diff
@@ -24,19 +24,18 @@ workflow LONG_READS_QC {
 
     ch_versions = ch_versions.mix(FASTP_LR.out.versions)
 
-    quality_levels_ch = FASTP_LR.out.json.map { meta, json -> {
-        json_txt = new JsonSlurper().parseText(json.text)
-        q20bases = json_txt?.summary?.before_filtering?.q20_bases ?: 0;
-        total_bases = json_txt?.summary?.before_filtering?.total_bases ?: 0;
-
-        q20_percentage = q20_bases / total_bases * 100
-
-        quality = [
-            "high_quality": q20_percentage >= 80, 
-            "low_quality": q20_percentage < 80,
-        ]
-        return [meta, quality]
-        } 
+    reads_json = FASTP_LR.out.reads.join( FASTP_LR.out.json )
+
+    reads_quality_levels = reads_json.map { meta, reads, json ->
+        def json_txt = new JsonSlurper().parseText(json.text)
+        
+        def q20_percentage = json_txt?.summary?.before_filtering?.q20_rate ?: 0;
+
+        if ( q20_percentage >= 0.8 ) {
```

Comment:
This should be a parameter which defaults to `0.8`

Reply:
This comes from Ales' benchmark of the quality threshold to distinguish pacbio hifi from non-hifi. It's the only wait to do it, as hifi is not standardised in the metadata. Should this parameter be customisable anyway?

Reply:
OK. Yeah, I would add it as parameter with the description too .. much easier to track and change in the future if needed